coderedcms/models/page_models.py

```diff
@@ -1053,3 +1069,247 @@ def serve_preview(self, request, mode):
+    )
+    latitude = models.FloatField(
+        blank=True,
+        null=True,
+        verbose_name=_("Latitude")
+    )
+    longitude = models.FloatField(
+        blank=True,
+        null=True,
+        verbose_name=_("Longitude")
+    )
+    auto_update_latlng = models.BooleanField(
+        default=True,
+        verbose_name=_("Auto Update Latitude and Longitude"),
+        help_text=_("If checked, automatically update the latitude and longitude when the address is updated.")
+    )
+    map_title = models.CharField(
+        blank=True,
+        max_length=255,
+        verbose_name=_("Map Title"),
+        help_text=_("If this is filled out, this is the title that will be used on the map.")
+    )
+    map_description = models.CharField(
+        blank=True,
+        max_length=255,
+        verbose_name=_("Map Description"),
+        help_text=_("If this is filled out, this is the description that will be used on the map.")
+    )
+    website = models.TextField(
+        blank=True,
+        verbose_name=_("Website")
+    )
+    phone_number = models.CharField(
+        blank=True,
+        max_length=255,
+        verbose_name=_("Phone Number")
+    )
+
+    content_panels = (
+        CoderedWebPage.content_panels[:1] + 
+        [
+            FieldPanel('address'),
+            FieldPanel('website'),
+            FieldPanel('phone_number'),
+        ] +
+        CoderedWebPage.content_panels[1:]
+    )
+
+    layout_panels = (
+        CoderedWebPage.layout_panels +
+        [
+            MultiFieldPanel(
+                [
+                    FieldPanel('map_title'),
+                    FieldPanel('map_description'),
+                ],
+                heading=_('Map Layout')
+            ),
+        ]
+    )
+
+    settings_panels = (
+        CoderedWebPage.settings_panels + 
+        [
+            MultiFieldPanel(
+                [
+                    FieldPanel('auto_update_latlng'),
+                    FieldPanel('latitude'),
+                    FieldPanel('longitude'),
+                ],
+                heading=_("Location Settings")
+            ),
+        ]
+    )
+
+    @property
+    def geojson_name(self):
+        return self.map_title or self.title
+
+    @property
+    def geojson_description(self):
+        return self.map_description
+
+    @property
+    def render_pin_description(self):
+        return render_to_string(
+            'coderedcms/includes/map_pin_description.html',
+            {
+                'name': self.geojson_name,
+                'address': self.address,
+                'url': self.url
+            }
+        )
+
+    @property
+    def render_list_description(self):
+        return render_to_string(
+            'coderedcms/includes/map_list_description.html',
+            {
+                'url': self.url,
+                'name': self.geojson_name,
+                'address': self.address,
+                'description': self.geojson_description
+            }
```

Comment:
One tiny change... please pass in `self` rather than the individual attributes, so devs would be able to customize the map_list_description template with whatever data they want.